File: clients/android/NewsBlur/src/com/newsblur/util/NotifySaveReceiver.java
package com.newsblur.util;

import com.newsblur.activity.Reading;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;

public class NotifySaveReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(final Context c, final Intent i) {
        final String storyHash = i.getStringExtra(Reading.EXTRA_STORY_HASH);
        NotificationUtils.cancel(c, storyHash.hashCode());
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... arg) {
                FeedUtils.offerInitContext(c);
                FeedUtils.dbHelper.putStoryDismissed(storyHash);
                FeedUtils.setStorySaved(storyHash, true, c);
                return null;
            }
        }.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
    }

}

File: clients/android/NewsBlur/app/src/main/java/com/newsblur/util/NotifySaveReceiver.kt
package com.newsblur.util

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.newsblur.activity.Reading
import com.newsblur.database.BlurDatabaseHelper
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class NotifySaveReceiver : BroadcastReceiver() {

    @Inject
    lateinit var dbHelper: BlurDatabaseHelper

    @Inject
    lateinit var feedUtils: FeedUtils

    override fun onReceive(c: Context, i: Intent) {
        val storyHash = i.getStringExtra(Reading.EXTRA_STORY_HASH)
        NotificationUtils.cancel(c, storyHash.hashCode())
        NBScope.executeAsyncTask(
                doInBackground = {
                    dbHelper.putStoryDismissed(storyHash)
                    feedUtils.setStorySaved(storyHash, true, c)
                }
        )
    }
}